File: python/sort/bubble_sort.py
def bubble_sort(a):
    for i in xrange(1, len(a) + 1):
        for j in xrange(len(a) - i):
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]
    return a

File: ruby/algorithms/sort/bubble_sort.rb
class Array
  def bubble_sort
    a = self
    (1..a.size).each do |i|
      (a.size - i).times do |j|
        if a[j] > a[j + 1]
          a[j], a[j + 1] = a[j + 1], a[j]
        end
      end
    end
    a
  end
end
